bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,276 @@
+                    break;
+                case SWINGLR_CHANNEL:
+                    uid = 6;
+                    value = Integer.parseInt(command.toString());
+                    break;
+                case TEMP_CHANNEL:
+                    uid = 9;
+                    value = (Integer.parseInt(command.toString().replace(" °C", ""))) * 10;
+                    break;
+            }
+        }
+        if (uid != 0) {
+            String contentString = "{\"command\":\"setdatapointvalue\",\"data\":{\"sessionID\":\"" + sessionId
+                    + "\", \"uid\":" + uid + ",\"value\":" + value + "}}";
+            String response = api.postRequest(ipAddress, contentString);
+            if (response != null && !response.isEmpty()) {
+                boolean success = IntesisHomeJSonDTO.getSuccess(response);
+                if (!success) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    String sessionString = "{\"command\":\"login\",\"data\":{\"username\":\"Admin\",\"password\":\""
+                            + password + "\"}}";
+                    response = api.postRequest(ipAddress, sessionString);
+                    if (response != null && !response.isEmpty()) {
+                        success = IntesisHomeJSonDTO.getSuccess(response);
+                        if (success) {
+                            JsonElement idNode = IntesisHomeJSonDTO.getData(response).get("id");
+                            AuthenticateData auth = gson.fromJson(idNode, AuthenticateData.class);
+                            sessionId = auth.sessionID;
+                            updateStatus(ThingStatus.ONLINE);
+                            response = api.postRequest(ipAddress, contentString);
+                            success = IntesisHomeJSonDTO.getSuccess(response);
+                            if (!success) {
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                            }
+                        } else {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        }
+                    }
+                } else {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+
+            }
+        }
+    }
+
+    /**
+     * Update device status and all channels
+     */
+    public void getAllUidValues() {
+        logger.debug("Polling IntesisHome device for actuall status");
+        if (thing.getStatus() != ThingStatus.ONLINE) {
+            String sessionString = "{\"command\":\"login\",\"data\":{\"username\":\"Admin\",\"password\":\"" + password
+                    + "\"}}";
+            String response = api.postRequest(ipAddress, sessionString);
+            if (response != null && !response.isEmpty()) {
+                boolean success = IntesisHomeJSonDTO.getSuccess(response);
+                if (success) {
```

Comment:
It might be a good approach to extract this code into a method and re-use it.

Reply:
Don't see the real benefit.